piq/feature_extractors/clip.py

```diff
@@ -0,0 +1,617 @@
+import hashlib
+import os
+import warnings
+
+import torch
+import torch.nn.functional as F
+
+from torch import nn
+from typing import Tuple, Union, Optional
+from collections import OrderedDict
+from urllib.request import urlopen
+
+
+CLIP_MODEL_PATH = (
+    "https://openaipublic.azureedge.net/clip/models/"
+    "afeb0e10f9e5a86da6080e35cf09123aca3b358a0c3e3b6c78a7b63bc04b6762/RN50.pt"
+)
+
+
+def _download(url: str, root: str) -> str:
+    r"""Downloads model's weights and caches them. If already downloaded - loads from cache.
+    Performs required SHA checksum verifications.
+
+    Args:
+        url: Web or file system path.
+        root: Absolute or relative path of the cache folder.
+
+    Returns:
+        Absolute or relative path of the model's weights.
+    """
+    os.makedirs(root, exist_ok=True)
+    filename = os.path.basename(url)
+
+    expected_sha256 = url.split("/")[-2]
+    download_target = os.path.join(root, filename)
+
+    if os.path.exists(download_target) and not os.path.isfile(download_target):
+        raise RuntimeError(f"{download_target} exists and is not a regular file")
+
+    if os.path.isfile(download_target):
+        if hashlib.sha256(open(download_target, "rb").read()).hexdigest() == expected_sha256:
+            return download_target
+        else:
+            warnings.warn(f"{download_target} exists, but the SHA256 checksum does not match; re-downloading the file")
+
+    with urlopen(url) as source, open(download_target, "wb") as output:
+        while True:
+            buffer = source.read(8192)
+            if not buffer:
+                break
+
+            output.write(buffer)
+
+    if hashlib.sha256(open(download_target, "rb").read()).hexdigest() != expected_sha256:
+        raise RuntimeError("Model has been downloaded but the SHA256 checksum does not not match")
+
+    return download_target
+
+
+def load() -> nn.Module:
```

Comment:
Function name is confusing, as it's not obvious what exactly it's "loading". Given that it takes 0 params let's rename it to `load_clip_model` or similar.

Reply:
As discussed, leave all features of the original implementation as is.